### DISCUSSION TITLE
Host dependency - How do I use a cmake-only dependency in my CMakeLists.txt?

### DISCUSSION THREAD

**Discussion by jsanchez19** [microsoft/vcpkg#20296] [Q&A]
I have a cross-compilation project that uses vcpkg in manifest mode with a dependency that has `"host": true`, but I'm unclear on how to use the CMake modules provided by this dependency in the CMakeLists.txt file of the project.

I tried `find_package(my_cool_cmake_utils)` but that fails. My port file doesn't provide a `config.cmake` file, so that makes sense. Ideally I would be able to do something like `include("my-cool-module.cmake")`, but that doesn't work either. I suppose that's because the `build/vcpkg_installed/x64-windows/share/my-cool-cmake-utils` directory is not in the `CMAKE_MODULE_PATH`. Should it be when I'm cross-compiling?

Below are what the `vcpkg.json` manifest in my project looks like, as well as the portfile of the dependency.

Project manifest file, vcpkg.json:
```json
{
  "name": "cross-compiled-embedded-project",
  "version": "0.0.1",
  "dependencies": [
    {"name": "my-cool-cmake-utils", "host": true}
  ]
}
```

The port for `my-cool-cmake-utils` looks like this (which is based on [`vcpkg-pkgconfig-get-modules`](https://github.com/microsoft/vcpkg/blob/1bb0fe61dfecd1450a1ff44c7fc8d424365366aa/ports/vcpkg-pkgconfig-get-modules/portfile.cmake):
```cmake
if(NOT TARGET_TRIPLET STREQUAL HOST_TRIPLET)
    # make FATAL_ERROR in CI when issue #16773 fixed
    message(WARNING "my-coolcmake-utils is a host-only port; please mark it as a host port in your dependencies.")
endif()

vcpkg_from_git(
    OUT_SOURCE_PATH SOURCE_PATH
    URL https://internal-git-host/my-cool-cmake.git
    REF 1234567890abcdef
)

file(
  INSTALL "${SOURCE_PATH}/LICENSE"
  DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"
  RENAME copyright)

file(COPY
    "${SOURCE_PATH}/cmake/my-cool-module.cmake"
    DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")

set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
```

**Comment by dg0yt** [comment]
AFAICT there is no supported way to determine the host triplet (and so the location of host tools etc.) from manifest mode. This limitation makes host dependencies somewhat pointless in manifest mode. Of course, you can manally adjust your environment variables (`PATH`) and cmake project configuration (`CMAKE_MODULE_PATH`).

This is the key blocker for easily using host tools (`yasm`, `msgmerge`, `pgkconf` etc.) in manifest mode.
(In port files, the host triplet is passed from the `vcpkg` tool to the script.) 

However, I would be very careful with adding Find modules built for host to the `CMAKE_MODULE_PATH` for the target.
